### ISSUE TITLE
Bitshift on usize fails depending on architecture

### ISSUE THREAD

**Issue by LukasKalbertodt** [rust-lang/rust#36269]
I'm not quite sure if this is a bug or intended behavior. 

```rust
let x = 1usize;
let y = x >> 34;
```

This compiles fine when compiling for `x86_64` but fails on 32 bit systems with:

```
error: bitshift exceeds the type's number of bits, #[deny(exceeding_bitshifts)] on by default
```

This really surprised me, as I never experienced this. The `From`/`Into` impls for `usize` also don't depend on the architecture, but assume the "worst" (namely, that `usize == u8`).

---

I expected to see this happen: an error in both cases or no error at all.

Instead, this happened: an error only for one architecture.

**Comment by nagisa** [comment]
Intended behaviour. `usize` is explicitly discouraged when unnecessary for various reasons, including the fact that its size is variable depending on target.
